docs/migration-guide-java.md

```diff
@@ -0,0 +1,157 @@
+
+## Migrating the code
+Consider this code which uses `raml-java-parser`:
+```java
+package co.acme.parse;
+
+import org.raml.v2.api.RamlModelBuilder;
+import org.raml.v2.api.RamlModelResult;
+import org.raml.v2.api.model.common.ValidationResult;
+import org.raml.v2.api.model.v10.api.Api;
+
+public class Raml10Parsing {
+  public static void parse() throws InterruptedException {
+    RamlModelResult ramlModelResult = new RamlModelBuilder().buildApi(input);
+    if (ramlModelResult.hasErrors()) {
+      for (ValidationResult validationResult : ramlModelResult.getValidationResults()) {
+        System.out.println(validationResult.getMessage());
+      }
+    } else {
+      Api api = ramlModelResult.getApiV10();
+    }
+  }
+}
+```
+
+Here's how it can be reworked to use `webapi-parser`:
+```java
+package co.acme.parse;
+
+import webapi.Raml10;
+import webapi.WebApiBaseUnit;
+import webapi.WebApiDocument;
+import amf.client.validate.ValidationReport;
+import amf.client.validate.ValidationResult;
+
+import java.util.concurrent.ExecutionException;
+
+public class Raml10Parsing {
+  public static void parse() throws InterruptedException, ExecutionException {
+    WebApiBaseUnit ramlModelResult = Raml10.parse(input).get();
+    ValidationReport validationReport = Raml10.validate(ramlModelResult).get();
+    if (!validationReport.conforms()) {
+      for (ValidationResult validationResult : validationReport.results()) {
+        System.out.println(validationResult.message());
+      }
+    } else {
+      WebApiDocument document = (WebApiDocument) ramlModelResult;
+    }
+  }
+}
+```
+
+In the example above, namespace `webapi` contains namespaces for all the supported API Spec formats: `Raml10`. `Raml08`, `Oas20`, `Oas30`, `AmfGraph`, each having an identical interface (OAS namespaces have an extra few methods). The list of supported operations each format supports includes parsing, resolution(flattening), validation and string generation.
+
+To get a description of each namespace and operation please research the new model API by following the link in the [assistance section](#need-assistance) below.
+
+## Detailed migration examples
+This section lists migration examples of the most common `raml-java-parser` parsing and model methods. Snippets are separated with a newline. First line of each example shows `raml-java-parser` method usage, while the second line shows how to achieve the same functionality with `webapi-parser` if possible. If no obvious alternative exists, a comment gives more detail.
+
+### Parsers
+```java
+import org.raml.v2.api.RamlModelBuilder;
+import webapi.Raml10;
+
+new RamlModelBuilder().buildApi(ramlFile);
```

Comment:
what does it do? I think we need to add a short description as comment before each block below

Reply:
Added comments.